common/taskana-common/src/main/java/pro/taskana/common/internal/util/ListUtil.java

```diff
@@ -0,0 +1,28 @@
+package pro.taskana.common.internal.util;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+public class ListUtil {
+
+  private ListUtil() {
+    throw new IllegalStateException("Utility class");
+  }
+
+  /**
+   * Splits a list with objects of type T into chunks of a certain size.
+   *
+   * @param <T> type of elements inside list
+   * @param inputList list to be divided
+   * @param size maximal number of elements inside chunk
+   * @return list containing the chunks
+   */
+  public static <T> Collection<List<T>> partitionBasedOnSize(Collection<T> inputList, int size) {
```

Comment:
`Collection<T> inputList` seems a little confusing ;)

Reply:
I'd name it `inputCollection`. But no need to change this here (you can do that in the next PR)

Reply:
Right, I do it on the next PR